lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -31,10 +46,26 @@ def client
     def perform_request(url, method, path, params={}, body=nil)
       params = (params || {}).merge @request_options
       params[:body] = body if body
-      # Convert URI object to string
-      url_and_path = path ? (url + path).to_s  : url.to_s
+      
+      # Create a new SafeURI that we can modify
+      request_uri = ::LogStash::Util::SafeURI.new(url.uri.clone)
+      
+      # Combine the paths using the minimal # of /s
+      # First, we make sure the path is relative so URI.join does
+      # the right thing
+      relative_path = path.start_with?("/") ? path[1..-1] : path
```

Comment:
we don't enforce path to be non null. somewhere up in the stack (with `healthcheck!`) this is causing exceptions to occur in the test run